jakarta-eclipse/src/org/jakarta/jdt/FilterDiagnosticsCollector.java

```diff
@@ -52,11 +57,40 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
 
 					
 					if (isWebFilterAnnotated && !isFilterImplemented) {
-						ISourceRange nameRange = JDTUtils.getNameRange(type);
-						Range range = JDTUtils.toRange(unit, nameRange.getOffset(), nameRange.getLength());
 						diagnostics.add(new Diagnostic(range, "Classes annotated with @WebFilter must implement the Filter interface."));
 					}
 					
+					/* URL pattern diagnostic check */
+					if(WebFilterAnnotation != null) {
+						IMemberValuePair[] memberValues = WebFilterAnnotation.getMemberValuePairs();
+						
+						boolean isUrlpatternSpecified = false;
+						boolean isServletNamesSpecified = false;
+						boolean isValueSpecified = false;
+						for (IMemberValuePair mv : memberValues) {
+							if(mv.getMemberName().equals(ServletConstants.URL_PATTERNS)) {
+								isUrlpatternSpecified = true;
+								continue;
+							}
+							if(mv.getMemberName().equals(ServletConstants.SERVLET_NAMES)) {
+								isServletNamesSpecified = true;
+								continue;
+							}
+							if(mv.getMemberName().equals(ServletConstants.VALUE)) {
+								isValueSpecified = true;
+							}
+							
+						}
+						
+						if (!isUrlpatternSpecified && !isValueSpecified && !isServletNamesSpecified) {
+							diagnostics.add(new Diagnostic(range, "The urlPatterns attribute, servletNames attribute or the value attribute of the annotation MUST be specified."));
+						}
+						if (isUrlpatternSpecified && isValueSpecified) {
+							diagnostics.add(new Diagnostic(range, "It is illegal to have both value and urlPatterns attribute used together on the same annotation."));
```

Comment:
Same comment as above, the range should be the range of the `@WebFilter` annotation, not the class name. 

Let's change the diagnostic message so that it is specific to WebFilter, something along the lines of:
`The WebFilter annotation cannot have both the 'value' and 'urlPatterns' attributes specified at once.`